src/components/CaptureFilter.js

```diff
@@ -367,22 +369,26 @@ function Filter(props) {
                 value={speciesId}
                 onChange={(e) => setSpeciesId(e.target.value)}
               >
-                {[
-                  { id: ALL_SPECIES, name: 'All' },
-                  {
-                    id: SPECIES_NOT_SET,
-                    name: 'Not set',
-                  },
-                  ...speciesContext.speciesList,
-                ].map((species) => (
-                  <MenuItem
-                    data-testid="species-item"
-                    key={species.id}
-                    value={species.id}
-                  >
-                    {species.name}
-                  </MenuItem>
-                ))}
+                {isLoading ? (
+                  <CircularProgress />
+                ) : (
+                  [
+                    { id: ALL_SPECIES, name: 'All' },
+                    {
+                      id: SPECIES_NOT_SET,
+                      name: 'Not set',
+                    },
+                    ...speciesContext.speciesList,
+                  ].map((species) => (
+                    <MenuItem
+                      data-testid="species-item"
```

Comment:
If there are multiple species, the data-testid will be the same for all of those species. Is that intended?

Reply:
I can't speak to the code here, I only added a ternary to render a loading animation or this code someone  else made.